File: code_samples/r/testing_mocking.r
# Testing: Mocking
# AI/ML Training Sample

Mocking <- setRefClass(
    "Mocking",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Mocking$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/go/testing_mocking.go
package main

import (
    "fmt"
)

// Testing: Mocking
// AI/ML Training Sample

type Mocking struct {
    Data string
}

func NewMocking() *Mocking {
    return &Mocking{
        Data: "",
    }
}

func (s *Mocking) Process(input string) {
    s.Data = input
}

func (s *Mocking) Validate() bool {
    return len(s.Data) > 0
}

func (s *Mocking) GetData() string {
    return s.Data
}

func main() {
    instance := NewMocking()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
